sourcegraph/sourcegraph
# Code Insights: Add mock code insight trend UI to the search result page

Part of https://github.com/sourcegraph/sourcegraph/issues/39228
Part of https://github.com/sourcegraph/sourcegraph/issues/39229

## Background

[Figma designs](https://www.figma.com/file/zJHFMF1RYta92C3Trp9y4b/Insights-on-the-search-screen-%2338443-Final-review?node-id=1887%3A10416)

This PR adds a new side panel tab for the search result page with "Simple UI" mode. This PR adds only UI part of the live preview project ([related Tracking issue here](https://github.com/sourcegraph/sourcegraph/issues/39421)) and all other work about connecting this UI with BE data will be addressed in [this issue](https://github.com/sourcegraph/sourcegraph/issues/39216). 

Note that all changes are hided behind the experimental feature flag `codeInsightsSearchResultTrends`. Also note that this UI is active only in "Simple UI" mode. 


## Test plan
- Activate "Simple UI mode" and go to the search result page
- Make sure that all UI on the page doesn't have any visual layout regressions
- Activate `codeInsightsSearchResultTrends` experimental feature flag. If you're looking at this PR with local build you can go to the [`experimentalFeatures.ts` file](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/web/src/stores/experimentalFeatures.ts?L19%3A34=&utm_product_name=WebStorm&utm_product_version=WebStorm&utm_source=JetBrains-v1.2.2) and add `codeInsightsSearchResultTrends: true` manually.
- Make sure that search result code insight page UI matches with original designs ([Figma designs](https://www.figma.com/file/zJHFMF1RYta92C3Trp9y4b/Insights-on-the-search-screen-%2338443-Final-review?node-id=1887%3A10416))
